ibis/expr/types/core.py

```diff
@@ -602,7 +602,11 @@ def unbind(self) -> ir.Table:
 
     def as_table(self) -> ir.Table:
         """Convert an expression to a table."""
-        raise NotImplementedError(type(self))
+        raise NotImplementedError("This expression cannot be converted to a table")
```

Comment:
Perhaps a more specific error message about what `This` is would be more useful.

Something like `"{type(self)} expressions cannot be converted into tables"`

Reply:
Updated.